tests/client/test_auth.py

```diff
@@ -50,6 +51,29 @@ async def request(
         return b"HTTP/1.1", self.status_code, b"", response_headers, response_stream
 
 
+class SyncMockTransport(httpcore.SyncHTTPTransport):
```

Comment:
I didn't manage to reuse the code from `AsyncMockTransport` without copy&pasting it.

Reply:
I would naively suggest moving the body of `request` to a separate function? Seems like it's all synchronous code, but it's not a big deal anyway.

Reply:
Personally I'm usually okay with a bit of carefully applied light duplication like this, but happy enough either ways. 😄

Reply:
I've moved it into a class instead, this way the `__init__` is also reused.